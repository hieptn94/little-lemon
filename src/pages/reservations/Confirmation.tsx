import { Link } from "@tanstack/react-router";
import { BookingInformation, ContactInformation } from "./schema";
import { Button } from "@/components/ui";

type Props = {
  bookingInformation: BookingInformation;
  contactInformation: ContactInformation;
};

function getPronoun(pronoun: ContactInformation["pronoun"]) {
  switch (pronoun) {
    case "mr": {
      return "Mr. ";
    }
    case "mrs": {
      return "Mrs. ";
    }
  }
}

export default function Confirmation({
  bookingInformation,
  contactInformation,
}: Props) {
  return (
    <section className="flex flex-col items-center text-dark-green-1 gap-6">
      <h1 className="font-bold text-2xl">Thank you!</h1>
      <p className="text-center">
        Dear {getPronoun(contactInformation.pronoun)}
        {contactInformation.name},<br />
        Your reservation for {bookingInformation.numberOfGuests} guests on{" "}
        {bookingInformation.date} at {bookingInformation.time} is confirmed.
      </p>
      <p className="text-center">
        We look forward to welcoming you to Little Lemon and hope you enjoy your
        meal.
      </p>
      <footer>
        <Link to="/">
          <Button variant="primary">Explore more</Button>
        </Link>
      </footer>
    </section>
  );
}
