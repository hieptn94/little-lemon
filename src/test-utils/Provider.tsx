import * as React from "react";
import { createRouter, RouterContextProvider } from "@tanstack/react-router";
import { routeTree } from "@/routeTree.gen";

export default function Provider({ children }: React.PropsWithChildren) {
  return (
    <RouterContextProvider
      router={createRouter({
        routeTree: routeTree,
      })}
    >
      {children}
    </RouterContextProvider>
  );
}
