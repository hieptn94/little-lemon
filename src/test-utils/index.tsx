import * as React from "react";
import { render, RenderOptions } from "@testing-library/react";
import Provider from "./Provider";

const customRender = (ui: React.ReactNode, options?: Omit<RenderOptions, "queries">) => {
  const actualOptions = options ?? {};
  return render(ui, { wrapper: Provider , ...actualOptions });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
