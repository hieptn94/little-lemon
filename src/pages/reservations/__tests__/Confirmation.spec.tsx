import { render } from "@/test-utils";
import Confirmation from "../Confirmation";

describe("<Confirmation />", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Confirmation
        bookingInformation={{
          date: "2024-08-11",
          time: "17:30",
          numberOfGuests: 4,
        }}
        contactInformation={{
          pronoun: "mr",
          name: "Alex",
          phone: "0123456789",
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
