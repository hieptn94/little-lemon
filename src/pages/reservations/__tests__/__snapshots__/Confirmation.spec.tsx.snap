// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Confirmation /> > should render correctly 1`] = `
<div>
  <section
    class="flex flex-col items-center text-dark-green-1 gap-6"
  >
    <h1
      class="font-bold text-2xl"
    >
      Thank you!
    </h1>
    <p
      class="text-center"
    >
      Dear 
      Mr. 
      Alex
      ,
      <br />
      Your reservation for 
      4
       guests on 
      2024-08-11
       
      at 
      17:30
       is confirmed.
    </p>
    <p
      class="text-center"
    >
      We look forward to welcoming you to Little Lemon and hope you enjoy your meal.
    </p>
    <footer>
      <a
        aria-current="page"
        class="active"
        data-status="active"
        href="/"
      >
        <button
          class="transition-colors rounded h-10 px-4 shadow-lg disabled:cursor-none bg-dark-green-1 text-white hover:bg-dark-green-2 active:bg-dark-green-3 disabled:bg-gray-50"
        >
          Explore more
        </button>
      </a>
    </footer>
  </section>
</div>
`;
